File: C++/programas/P-Dibujar figuras/P-Dibujar figuras/Figuras.cpp
#include <iostream> //Cabecera para entrada y salida de datos.
//#include <cstdio>
#include "Figuras.h" //Cabecera de clase Figuras.

//Procedimiento que dibuja un triángulo con asteriscos.
void Figuras::DibujarTriangulo() {
	short Altura, R, C, B;

	system("cls"); //Limpia la terminal.

	//Ingreso de valor a variable
	std::cout << "Altura del triángulo (número entero): "; std::cin >> Altura;
	if (std::cin.fail()) //Si se ingresó un tipo de dato incorrecto en la variable.
	{
		std::cout << "Error de formato. ";
		system("pause"); //Pausa el flujo del programa.
		std::cin.clear(); //Limpia el buffer de cin.
		//Ignora el resto de caracteres ingresados en cin hasta el próximo salto de línea.
		std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
		DibujarTriangulo(); //Llamada recursiva a DibujarTriangulo.
	}
	else
	{
		std::cout << "Triángulo:\n\n";
		//Dibuja el triángulo.
		for (R = 1; R <= Altura; R++) { //Bucle para la altura.
			for (B = 1; B <= (Altura - R); B++) { //Bucle para los espacios en blanco.
				std::cout << " ";
			}

			for (C = 1; C <= (R * 2) - 1; C++) { //Bucle para los asteriscos que conforman al triángulo.
				std::cout << "*";
			}
			std::cout << "\n";
		}

		system("pause"); //Pausa el flujo del programa.
	}
}

//Procedimiento que dibuja un triángulo con asteriscos.
void Figuras::DibujarRectangulo() {
	short Base, Altura, R, C;

	system("cls"); //Limpia la terminal.

	//Ingreso de datos a variables.
	std::cout << "Altura del rectángulo (número entero): "; std::cin >> Altura;
	std::cout << "Base del rectángulo (número entero): "; std::cin >> Base;

	if (std::cin.fail()) //Si se ingresó un tipo de dato incorrecto en la variable.
	{
		std::cout << "Error de formato. ";
		system("pause"); //Pausa el flujo del programa.
		std::cin.clear(); //Limpia el buffer de cin.
		//Ignora el resto de caracteres ingresados en cin hasta el próximo salto de línea.
		std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
		DibujarRectangulo(); //Llamada recursiva a DibujarRectangulo.
	}
	else
	{
		std::cout << "Rectángulo:\n\n";
		//Dibuja el rectángulo
		for (R = 1; R <= Altura; R++) { //Bucle para la altura
			for (C = 1; C <= Base; C++) { //Bucle para la anchura
				std::cout << "*  ";
			}
			std::cout << "\n";
		}

		system("pause"); //Pausa el flujo del programa.
	}
}

//Procedimiento que dibuja un triángulo con asteriscos.
void Figuras::DibujarCuadrado() {
	short Lado, R, C;

	system("cls"); //Limpia la term
	
	//Ingreso de dato a variable.
	std::cout << "Lado del cuadrado (número entero): "; std::cin >> Lado;

	if (std::cin.fail()) //Si se ingresó un tipo de dato incorrecto en la variable.
	{
		std::cout << "Error de formato. ";
		system("pause"); //Pausa el flujo del programa.
		std::cin.clear(); //Limpia el buffer de cin.
		//Ignora el resto de caracteres ingresados en cin hasta el próximo salto de línea.
		std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
		DibujarCuadrado(); //Llamada recursiva.
	}
	else
	{
		std::cout << "Cuadrado:\n\n";
		//Dibuja el cuadrado.
		for (R = 1; R <= Lado; R++) { //Bucle para la un lado
			for (C = 1; C <= Lado; C++) { //Bucle para un lado
				std::cout << "* ";
			}
			std::cout << "\n";
		}

		system("pause"); //Pausa el flujo del programa.
	}
}

File: Java/programas/Dibujar Figuras/src/Figuras.java
import java.io.IOException; //Excepción si hay error durante entrada y salida de datos.
import java.util.InputMismatchException; //Excepción si hay un error de formato en tipos de datos.
import java.util.Scanner; //Librería para declara objeto Scanner de entrada y salida.

public class Figuras {
    private Scanner lec = new Scanner(System.in); //Declara objeto Scanner.

    //Procedimiento que dibuja un triángulo en base a la altura del mismo.
    public void DibujarTriangulo() throws IOException{
        byte Altura, R, C, B;

        //Selectiva try-catch por si se ingresa un tipo de dato incorrecto en la variable.
        try{
            System.out.printf("Altura del triángulo (número entero): "); Altura = lec.nextByte();

            System.out.println("Triángulo:\n"); //Bucles que imprimen un triángulo con una altura asignada por el usuario.
            for (R = 1; R <= Altura; R++) { //Bucle para la altura.
                for (B = 1; B <= (Altura - R); B++) { //Bucle para los espacios en blanco.
                    System.out.printf(" ");
                }
    
                for (C = 1; C <= (R * 2) - 1; C++) { //Bucle para los asteriscos que conforman al triángulo.
                    System.out.printf("*");
                }
                System.out.printf("\n"); //Salto de línea que da forma al triángulo.
            }
            System.out.printf("Presiona cualquier tecla para ir al menú.");
            System.in.read(); //Espera a que se presione ENTER para continuar.
            
        } catch(InputMismatchException e){
            System.out.print("Error de formato.\n");
            lec.nextLine(); //Limpia buffer de entrada de datos.
            DibujarTriangulo(); //Llamada recursiva a DibujarTriangulo.
        }
    }

    //Procedimiento que dibuja un rectángulo según su base y altura.
    public void DibujarRectangulo() throws IOException{
        byte Base, Altura, R, C;

        //Selectiva try-catch por si se ingresa un tipo de dato incorrecto en la variable.
        try{
            System.out.printf("Base del rectángulo (número entero): "); Base = lec.nextByte();
            System.out.printf("Altura del rectángulo (número entero): "); Altura = lec.nextByte();
            
            System.out.println("Rectángulo:\n"); //Bucles que imprimen un rectángulo con una base y altura asignada por el usuario.
            for (R = 1; R <= Altura; R++) { //Bucle para la altura.
                for (C = 1; C <= Base; C++){ //Bucle para la anchura.
                    System.out.print("*  ");
                }
                System.out.printf("\n"); //Salto de línea que da forma al triángulo.
            }
            System.out.printf("Presiona cualquier tecla para ir al menú.");
            System.in.read(); //Espera a que se presione ENTER para continuar.
            
        } catch(InputMismatchException e){
            System.out.print("Error de formato.\n");
            lec.nextLine(); //Limpia buffer de entrada de datos.
            DibujarRectangulo(); //Llamada recursiva a DibujarTriangulo.
        }
    }

    //Procedimiento que dibuja un cuadrado según la medida de su lado.
    public void DibujarCuadrado() throws IOException{
        byte Lado, R, C;

        //Selectiva try-catch por si se ingresa un tipo de dato incorrecto en la variable.
        try{
            System.out.printf("Lado del cuadrado (número entero): "); Lado = lec.nextByte();
            
            System.out.println("Cuadrado:\n"); //Bucles que imprimen un cuadrado con una medida de lado asignada por el usuario.
            for (R = 1; R <= Lado; R++) { //Bucle para la un lado.
                for (C = 1; C <= Lado; C++){ //Bucle para un lado.
                    System.out.print("* ");
                }
                System.out.printf("\n"); //Salto de línea que da forma al cuadrado.
            }
            System.out.printf("Presiona cualquier tecla para ir al menú.");
            System.in.read(); //Espera a que se presione ENTER para continuar.
            
        } catch(InputMismatchException e){
            System.out.print("Error de formato.\n");
            lec.nextLine(); //Limpia buffer de entrada de datos.
            DibujarCuadrado(); //Llamada recursiva a DibujarCuadrado.
        }
    }
}
